core/src/main/java/io/ddf/content/ViewHandler.java

```diff
@@ -137,13 +137,8 @@ public DDF removeColumns(List<String> columnNames) throws DDFException {
     }
 
     DDF newddf = this.project(columns);
-    if(this.getDDF().isMutable()) {
-      this.getDDF().updateInplace(newddf);
-      return this.getDDF();
-    } else {
-      newddf.getMetaDataHandler().copyFactor(this.getDDF());
-      return newddf;
-    }
+    newddf.getMetaDataHandler().copyFactor(this.getDDF());
```

Comment:
I mean this PR said it just change `castType` and `transform`, not `removeColumns`
This change will make removeColumns return new DDF instead of the same DDF as of now